### PR TITLE
dev-java/ant-contrib: bump to 1.0_beta6_pre20201123

### PR DESCRIPTION
**Pull request by vaukai**
Package-Manager: Portage-3.0.17, Repoman-3.0.2
Signed-off-by: Volkmar W. Pogatzki <gentoo@pogatzki.net>

### CODE REVIEW

**Comment by vaukai** [comment]
`dev-java/ant-ivy` is complicated.  Does not compile, too many errors.
Added because MAVEN_ID of 2.3.0 was not accepted by `java-ebuilder`.

For the main topic, `dev-java/ant-contrib`, there are still unmatched dependencies.
Can we add two `MAVEN_ID`s to a single ebuild, 
one for `org.apache.ant:ant:1.9.15` and 
one for `org.apache.ant:ant-launcher:1.9.5`?

And where to find `org.jvnet.hudson:ivy:1.4.1`?

File: dev-java/ant-contrib/ant-contrib-1.0_beta6_pre20201123.ebuild [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+SRC_URI="https://github.com/cniweb/${PN}/archive/${MY_COMMIT}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="Apache-2.0"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc64 ~x86"
+
+# Common dependencies
+# POM: pom.xml
+# commons-httpclient:commons-httpclient:3.1 -> >=dev-java/commons-httpclient-3.1:3
+# org.apache.ant:ant:1.9.15 -> !!!groupId-not-found!!!
```
Comment (fordfrog) [inline dev-java/ant-contrib/ant-contrib-1.0_beta6_pre20201123.ebuild:27]: i think dependency on `ant-core` might work.

File: dev-java/ant-contrib/ant-contrib-1.0_beta6_pre20201123.ebuild [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+#		>=dev-java/ant-ivy-2.5.0:2
+#		>=dev-java/bcel-6.5.0:0
+#		>=dev-java/commons-httpclient-3.1:3
+#		>=dev-java/xerces-2.12.0:2
+#	"
+
+# Compile dependencies
+# POM: pom.xml
+# test? junit:junit:4.12 -> >=dev-java/junit-4.12:4
+# test? org.apache.ant:ant-launcher:1.9.5 -> !!!groupId-not-found!!!
```
Comment (fordfrog) [inline dev-java/ant-contrib/ant-contrib-1.0_beta6_pre20201123.ebuild:44]: `ant-launcher` is a part of `ant-core` so the dependency above should be enough.

File: dev-java/ant-contrib/ant-contrib-1.0_beta6_pre20201123.ebuild [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+SLOT="0"
+KEYWORDS="~amd64 ~ppc64 ~x86"
+
+# Common dependencies
+# POM: pom.xml
+# commons-httpclient:commons-httpclient:3.1 -> >=dev-java/commons-httpclient-3.1:3
+# org.apache.ant:ant:1.9.15 -> !!!groupId-not-found!!!
+# org.apache.bcel:bcel:5.1 -> >=dev-java/bcel-6.5.0:0
+# org.apache.ivy:ivy:2.5.0 -> >=dev-java/ant-ivy-2.5.0:2
+# org.jvnet.hudson:ivy:1.4.1 -> !!!groupId-not-found!!!
```
Comment (fordfrog) [inline dev-java/ant-contrib/ant-contrib-1.0_beta6_pre20201123.ebuild:30]: you might try to ignore this dep and check whether not including it really causes any issue.

**Comment by fordfrog** [comment]
> `dev-java/ant-ivy` is complicated. Does not compile, too many errors.
> Added because MAVEN_ID of 2.3.0 was not accepted by `java-ebuilder`.

`java-ebuilder` provides just a hint based on content of `pom.xml`. you might try to go with another version of the dep (like in this case the one we have) and see whether it causes any issues or not.

> For the main topic, `dev-java/ant-contrib`, there are still unmatched dependencies.
> Can we add two `MAVEN_ID`s to a single ebuild,
> one for `org.apache.ant:ant:1.9.15` and
> one for `org.apache.ant:ant-launcher:1.9.5`?

i think in our case `org.apache.ant:ant:1.9.15` should match to our `ant-core` in fact. and the same goes for `org.apache.ant:ant-launcher:1.9.5`. also, not necessarily we need to have every package resolved, some human touches might be still needed.

> And where to find `org.jvnet.hudson:ivy:1.4.1`?

as mentioned in the ebuild, try it without it. i did a quick check but did not find it. afaik, hudson was a continuous integration tool which is now discontinued, see [here](https://en.wikipedia.org/wiki/Hudson_(software)).

**Comment by vaukai** [comment]
> > `dev-java/ant-ivy` is complicated. Does not compile, too many errors.
> > Added because MAVEN_ID of 2.3.0 was not accepted by `java-ebuilder`.
> 
> `java-ebuilder` provides just a hint based on content of `pom.xml`. you might try to go with another version of the dep (like in this case the one we have) and see whether it causes any issues or not.

With `dev-java/ant-ivy-2.3.0-r3` I get
```
 * Compiling ...
warning: [options] bootstrap class path not set in conjunction with -source 8
src/main/java/net/sf/antcontrib/net/Ivy14Adapter.java:11: error: package fr.jayasoft.ivy.ant does not exist
import fr.jayasoft.ivy.ant.IvyCacheFileset;
                          ^
src/main/java/net/sf/antcontrib/net/Ivy14Adapter.java:12: error: package fr.jayasoft.ivy.ant does not exist
import fr.jayasoft.ivy.ant.IvyConfigure;
                          ^
src/main/java/net/sf/antcontrib/net/Ivy14Adapter.java:17: error: cannot find symbol
                IvyConfigure configure = new IvyConfigure();
                ^
  symbol:   class IvyConfigure
  location: class Ivy14Adapter
src/main/java/net/sf/antcontrib/net/Ivy14Adapter.java:17: error: cannot find symbol
                IvyConfigure configure = new IvyConfigure();
                                             ^
  symbol:   class IvyConfigure
  location: class Ivy14Adapter
src/main/java/net/sf/antcontrib/net/Ivy14Adapter.java:104: error: cannot find symbol
                IvyCacheFileset cacheFileSet = new IvyCacheFileset();
                ^
  symbol:   class IvyCacheFileset
  location: class Ivy14Adapter
src/main/java/net/sf/antcontrib/net/Ivy14Adapter.java:104: error: cannot find symbol
                IvyCacheFileset cacheFileSet = new IvyCacheFileset();
                                                   ^
  symbol:   class IvyCacheFileset
  location: class Ivy14Adapter
```

File: dev-java/ant-contrib/ant-contrib-1.0_beta6_pre20201123.ebuild (hunk 2 above)
Comment (vaukai) [reply dev-java/ant-contrib/ant-contrib-1.0_beta6_pre20201123.ebuild]: Yep, setting available `ant-core` for both seems to work

**Comment by fordfrog** [comment]
@vaukai for some reason, our `ant-ivy:2` doesn't have that class in package `fr.jayasoft.ivy.ant` but instead in package `org.apache.ivy.ant`. patching the file `src/main/java/net/sf/antcontrib/net/Ivy14Adapter.java` and replacing all `import fr.jayasoft.ivy.ant.*` with `import org.apache.ivy.ant.*` should fix this issue.

**Comment by vaukai** [comment]
> @vaukai for some reason, our `ant-ivy:2` doesn't have that class in package `fr.jayasoft.ivy.ant` but instead in package `org.apache.ivy.ant`. patching the file `src/main/java/net/sf/antcontrib/net/Ivy14Adapter.java` and replacing all `import fr.jayasoft.ivy.ant.*` with `import org.apache.ivy.ant.*` should fix this issue.

Not too bad.  It compiles and for tests it needs some adjustment

**Comment by fordfrog** [comment]
eh, i accidentally merged this commit...

**Comment by vaukai** [comment]
> eh, i accidentally merged this commit...

No problem.  Test can still be fixed later.  Some test bot will come

**Comment by vaukai** [comment]
All testing trouble starts in `test/resources/antclipse/antclipsetest.xml` where I don't understand
```
    34         <!-- makes a zip file with all your sources (supposing you have only source directory) -->
    35          <zip destfile="src.zip"><fileset refid="ecl2-${ecl2srcpath}"/></zip>
```
What is `${ecl2srcpath}`? How could I check?

**Comment by fordfrog** [comment]
i saw somewhere in the failing tests that it wants to search for some sources but those are probably missing from the package

**Comment by vaukai** [comment]
Should I start excluding those tests? Or will you simply put `RESTRICT="test"`?

**Comment by fordfrog** [comment]
i'm currently adding tests to `bcel` to verify it really works as it should. still 7 tests failing, but i will have to go to do something else. will commit it, a bug will be triggered, maybe we get it fixed it later.

**Comment by fordfrog** [comment]
here's the commit: 92b8c2bfa58dd8bb664724ce8413374b2131dfbc
it seems `bcel`'s method for retrieval of method variables does not work as it should.

**Comment by fordfrog** [comment]
just checked with the upstream sources using `mvn clean test` and it fails too:
```
[INFO] Running org.apache.bcel.generic.MethodGenTestCase
[ERROR] Tests run: 5, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 0.002 s <<< FAILURE! - in org.apache.bcel.generic.MethodGenTestCase
[ERROR] org.apache.bcel.generic.MethodGenTestCase.testRemoveLocalVariable  Time elapsed: 0 s  <<< FAILURE!
org.junit.ComparisonFailure: variable name expected:<[a]> but was:<[this]>
	at org.apache.bcel.generic.MethodGenTestCase.testRemoveLocalVariable(MethodGenTestCase.java:84)

[ERROR] org.apache.bcel.generic.MethodGenTestCase.testRemoveLocalVariables  Time elapsed: 0 s  <<< FAILURE!
org.junit.ComparisonFailure: variable name expected:<[a]> but was:<[this]>
	at org.apache.bcel.generic.MethodGenTestCase.testRemoveLocalVariables(MethodGenTestCase.java:106)
```
though they have just 2 failures, we have 7...

**Comment by fordfrog** [comment]
@vaukai one more hint (not related to this bug). when touching an ebuild, it's good to try build the package against the latest slot. `snakeyaml` builds fine against latest `commons-lang`, i updated that one (you probably noticed that nattka flood).

**Pull request closed without merging**